frontend/src/service/defenceService.ts

```diff
@@ -54,15 +57,21 @@ async function configureDefence(
 	defenceId: DEFENCE_ID,
 	config: DefenceConfigItem[],
 	level: LEVEL_NAMES
-): Promise<boolean> {
+): Promise<ChatMessage | null> {
 	const response = await sendRequest(`${PATH}configure`, {
 		method: 'POST',
 		headers: {
 			'Content-Type': 'application/json',
 		},
 		body: JSON.stringify({ defenceId, config, level }),
 	});
-	return response.status === 200;
+
+	const { resultingChatInfoMessage } =
+		(await response.json()) as ConfigureDefenceResponse;
```

Comment:
This could be problematic because the response type in the case of errors is not application/json, it's text/plain. So I expect `response.json()` to fail in that case. Did you try it?

Better to check the status first, and if Ok then parse the JSON and make your DTO. And if it's not Ok, what happens to the error message? Looks like we are swallowing that currently.

Reply:
Yup, good call to check the status first. And yes, the message gets swallowed.

Reply:
not-swallowing the errors is a wider UI problem I reckon

Reply:
It's that "unhappy paths" issue come back to bite us 😁